src/metrics/site_level.jl

```diff
@@ -86,12 +86,26 @@ YAXArray with summary metric for the remaining axis.
 function summarize(
     data::YAXArray{D,T,N,A}, alongs_axis::Vector{Symbol}, metric::Function
 )::YAXArray where {D,T,N,A}
+    # The approach using JuliennedArrays is faster then directly passing the YAXArray to
+    # mapslices only when we `read(data)`. Since `read(data)` loads all the data from disk
+    # into memory, we only want to use that approach when there is a reasonable amount of
+    # available space in RAM
+    data_size = Base.summarysize(data)
+    free_ram = Sys.free_memory()
+    proportional_usage = data_size / free_ram
+
+    # Only use this approach when data occupies more than 70% of available space in RAM
+    if proportional_usage > 0.7 * free_ram
+        # `D.` is ensuring the returned YAXArray has the same type as the input `data`
+        return D.(mapslices(metric, data; dims=alongs_axis))
+    end
```

Comment:
Suggesting a more readable if statement

```suggestion
    data_size = Base.summarysize(data)
    
    # Only use this approach when data occupies more than 70% of available space in RAM
    available_threshold = Sys.free_memory() * 0.7
    proportional_usage = data_size / free_ram
    if proportional_usage > available_threshold 
        # `D.` is ensuring the returned YAXArray has the same type as the input `data`
        return D.(mapslices(metric, data; dims=alongs_axis))
    end
```